<?xml version="1.0"?>
<launch>
    <!-- Start Gazebo with game world -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find cs_bot)/worlds/game_world.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!-- T team robot -->
    <include file="$(find cs_bot)/launch/single_robot_no_gazebo.launch">
        <arg name="robot_name" value="t_robot1"/>
        <arg name="x_pos" value="0.0"/>
        <arg name="y_pos" value="0.0"/>
        <arg name="team_color" value="red"/>
    </include>

    <!-- CT team robot -->
    <include file="$(find cs_bot)/launch/single_robot_no_gazebo.launch">
        <arg name="robot_name" value="ct_robot1"/>
        <arg name="x_pos" value="0.0"/>
        <arg name="y_pos" value="5.0"/>
        <arg name="team_color" value="blue"/>
    </include>

    <!-- T team robot controller -->
    <node pkg="cs_bot" type="robot_controller.py" name="t_robot1_controller" output="screen">
        <param name="robot_name" value="t_robot1"/>
        <param name="team" value="T"/>
        <param name="weapon" value="RIFLE"/>
    </node>

    <!-- CT team robot controller -->
    <node pkg="cs_bot" type="robot_controller.py" name="ct_robot1_controller" output="screen">
        <param name="robot_name" value="ct_robot1"/>
        <param name="team" value="CT"/>
        <param name="weapon" value="SNIPER"/>
    </node>

    <!-- Game manager node -->
    <node pkg="cs_bot" type="game_manager.py" name="game_manager" output="screen"/>

    <!-- Start RViz visualization -->
    <node name="rviz" pkg="rviz" type="rviz" 
          args="-d $(find cs_bot)/rviz/test_config.rviz" />
</launch>
